Orange/widgets/unsupervised/owdistances.py

```diff
@@ -144,7 +157,7 @@ def compute_distances(self, metric, data):
                 self.Error.mahalanobis_error(e)
                 return
 
-        return metric(data, data, 1 - self.axis, impute=True)
+        return compute(metric, data)
```

Comment:
A stylistic comment. I don't like the obvious radon fixes --- here's a code that's too complex so let's tear out a chunk of it into a separate function.

The main job of `compute_distances` is not to run various checks and fixes, and in the end, yes, call a local function that computes distances.

Turn it around. The "auxiliary" code should be the code that prepares the data and not the code that does the work. So, factor out the preprocessing/checking instead.